pkg/core/namespace.go

```diff
@@ -210,30 +299,42 @@ func (c *client) initServiceAccount(options *NamespaceInitOptions, initLabels ma
 	if err != nil {
 		return err
 	}
-	if options.secretType() != secretTypeNone {
+	if options.initSecretType() != secretTypeNone {
 		secretName := options.SecretName
-		secretAlreadyPresent := false
-		for _, s := range serviceAccount.Secrets {
-			if s.Name == secretName {
-				secretAlreadyPresent = true
-				break
-			}
-		}
-		if secretAlreadyPresent {
-			fmt.Printf("Serviceaccount %q already exists in namespace %q with secret %q. Skipping.\n", serviceAccount.Name, ns, secretName)
-		} else {
-			serviceAccount.Secrets = append(serviceAccount.Secrets, corev1.ObjectReference{Name: secretName})
-			fmt.Printf("Adding secret %q to serviceaccount %q in namespace %q\n", secretName, serviceAccount.Name, ns)
-			_, err = c.kubeClient.CoreV1().ServiceAccounts(ns).Update(serviceAccount)
-			if err != nil {
-				return err
-			}
+		return c.updateServiceAccount(ns, secretName, serviceAccount)
+	}
+	return nil
+}
+
+func (c *client) updateServiceAccount(namespace, secretName string, serviceAccount *corev1.ServiceAccount) error {
+	secretAlreadyPresent := false
+	for _, s := range serviceAccount.Secrets {
+		if s.Name == secretName {
+			secretAlreadyPresent = true
+			break
 		}
 	}
+	if secretAlreadyPresent {
+		fmt.Printf("Secret %q is already bound to serviceaccount %q in namespace %q. Skipping.\n", secretName, serviceAccount.Name, namespace)
```

Comment:
Do we need this? The message is more likely to confuse users than inform them.

Reply:
It's the same logic as for `namespace init`, I just rephrased it to make it clearer.

Reply:
Fixed in newest commit.